Automattic/wp-calypso
# Global plugins marketplace - fix layout breakage due to pugins browser container and section component.

Related to # Automattic/dotcom-forge#8272

## Proposed Changes

* Removes these`::before` pseudo element styles from the `.plugins-browser__main-container` and `<Section>` component used by the Marketplace Footer while at the global level.
    * The former is removed by changing the SCSS selector from `&::before` to  `.theme-default:not(.is-global) &::before`.
    * The latter style is applied via emotion/styled, which doesn't seem to support parent element selector chains like the above example. So here I have had to add an extra property to the section component and its styled component. The `hideBackgroundElement` prop ensures we do not apply the pseudo styles, and we apply this at the `MarketplaceFooter` component level based on the `getShouldShowGlobalSidebar` selector.


BEFORE - content frame loses top padding and gains bottom padding once loading into the plugin (and back at the main list as well)

![global-plugin-layout-break](https://github.com/user-attachments/assets/29bb9271-052c-494e-8e92-81b1f437468b)


AFTER - content frame is stable.

![global-plugins-good](https://github.com/user-attachments/assets/e7cec3a7-6383-4662-a8bb-cf75eec71b74)


NOTES

1. It looks like these pseudo elements were used to apply background colors to these sections, outside of the margins they are constrained to. This absolute positioning seems to break the layout of the global frame. They are not applying any background in the global area currently either. We can just remove them from there.
2. Im not thrilled about needing to add extra props here. I would like to just restrict via parent selectors like in the SCSS file, but the styled components do not seem to support this? I may be missing a way to do this from the styled component instead.
3. Im not 100% sure why this absolute positioning breaks the vertical layout of the content frame. But it clearly seems to be the culprit.

## Why are these changes being made?


* There is a bug at the global plugins marketplace where the layout breaks when going back and forth between the plugins list and individual plugins pages. This seems to be due to the absolutely positioned pseudo elements. This seems to happen only after the first render as it does not appear on load, but once these elements are later added or removed from the layout.



## Testing Instructions



* Go to the global view
* Select plugins
* Click into a plugin (I am using yoast as seen in the video above)
    * I noticed in testing this not all individual plugin pages seem to cause this. So on production, find a plugin such that when you click into it the above layout breakage is observed. Use this plugin in testing this PR.
* Verify the layout is not broken.
* Go back to the main plugins broswer. Verify the layout is still not broken.
* Visit the plugins page while logged out. Verify there are no changes from production.
* Visit the individual site level plugins page (the site sidebar visible at my home -> plugins). Verify there are no changes from production.

## Pre-merge Checklist
